src/filter/filetypes.rs

```diff
@@ -0,0 +1,56 @@
+use crate::filesystem;
+use crate::walk;
+
+use super::Filter;
+
+/// Whether or not to show
+#[derive(Clone)]
+pub struct FileTypes {
+    pub files: bool,
+    pub directories: bool,
+    pub symlinks: bool,
+    pub sockets: bool,
+    pub pipes: bool,
+    pub executables_only: bool,
+    pub empty_only: bool,
+}
+
+impl Default for FileTypes {
+    fn default() -> FileTypes {
+        FileTypes {
+            files: false,
+            directories: false,
+            symlinks: false,
+            sockets: false,
+            pipes: false,
+            executables_only: false,
+            empty_only: false,
+        }
+    }
+}
+
+impl Filter for FileTypes {
+    fn should_skip(&self, entry: &walk::DirEntry) -> bool {
+        entry
+            .file_type()
+            .map(|ref entry_type| {
+                (!self.files && entry_type.is_file())
+                    || (!self.directories && entry_type.is_dir())
+                    || (!self.symlinks && entry_type.is_symlink())
+                    || (!self.sockets && filesystem::is_socket(*entry_type))
+                    || (!self.pipes && filesystem::is_pipe(*entry_type))
+                    || (self.executables_only
+                        && !entry
+                            .metadata()
+                            .map(|m| filesystem::is_executable(&m))
```

Comment:
Can we please fix the two clippy annotations here and below?

Reply:
Oh, am I running the command incorrectly? When I run `cargo clippy` on the code I don't get any warnings at all.

Reply:
You might simply have an older version of the Rust toolchain where this clippy check is not yet supported? You can also try to use `cargo clippy --all-targets --all-features`, but that shouldn't be necessary here I think.

Reply:
Yep, had an older version; I can see the warnings now. I'll handle those too.